edge-util/src/Microsoft.Azure.Devices.Edge.Storage/DbStoreDecorator.cs

```diff
@@ -0,0 +1,106 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace Microsoft.Azure.Devices.Edge.Storage
+{
+    using System;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Microsoft.Azure.Devices.Edge.Util;
+
+    /// <summary>
+    /// A decorating wrapper over a Db store to provide ease of extending DB store behavior.
+    /// </summary>
+    public abstract class DbStoreDecorator : IDbStore
```

Comment:
Should we be using this on our other DB stores as well? (ColumnFamilyDbStore, DiskSpaceAware once it goes in)

Reply:
Yeah, I think you should for DiskSpaceAware